app/controllers/search_controller.rb

```diff
@@ -30,13 +30,27 @@ def extract_errors(response)
   end
 
   def extract_facets(response)
-    response&.data&.search&.to_h&.dig('aggregations')
+    facets = response&.data&.search&.to_h&.dig('aggregations')
+
+    facets&.map { |k, v| { facet_labels[k] => v } }
   end
 
   def extract_results(response)
     response&.data&.search&.to_h&.dig('records')
   end
 
+  def facet_labels
```

Comment:
Non-blocking: this feels more like a Helper or Model concern than Controller method.